testing/databaseTest.py

```diff
@@ -294,10 +294,19 @@ def kinetics_checkCorrectNumberofNodesInRules(self, family_name):
         """
         family = self.database.kinetics.families[family_name]
         expectedNumberNodes = len(family.getRootTemplate())
+        tst = []
         for label, entries in family.rules.entries.iteritems():
             for entry in entries:
                 nodes = label.split(';')
-                nose.tools.assert_equal(len(nodes), expectedNumberNodes, "Wrong number of groups or semicolons in family {family} rule {entry}.  Should be {num_nodes}".format(family=family_name, entry=entry, num_nodes=expectedNumberNodes))
+                tst.append((len(nodes), expectedNumberNodes, "Wrong number of groups or semicolons in family {family} rule {entry}.  Should be {num_nodes}".format(family=family_name, entry=entry, num_nodes=expectedNumberNodes)))
+
+        boo = False
+        for item in tst:
+            if item[0] != item[1]:
```

Comment:
For all of these tests, my initial thought is that the check should occur when you're iterating through the entries, and you should just store a list of error messages that need to be printed. Then at the end (here), if the list is not empty, log all of the stored messages and fail the test.

It seems cleaner that way. Let me know what you think.